phoenix-core/src/main/java/org/apache/phoenix/iterate/ExplainTable.java

```diff
@@ -178,10 +184,19 @@ protected void explain(String prefix, List<String> planSteps,
                 }
             } while (filterIterator.hasNext());
         }
+        String whereFilterStr = null;
         if (whereFilter != null) {
+            whereFilterStr = whereFilter.toString();
+        } else {
+            byte[] expBytes = scan.getAttribute(BaseScannerRegionObserver.LOCAL_INDEX_FILTER_STR);
```

Comment:
Expression.readFields does *not* restore all display properties. Since I cannot change how expressions are (de) serialized I came up with this.
A bit unfortunate that this has to be on the scan and is uselessly also sent to the server - although it's helpful there for debugging.